skopt/gbrt_opt.py

```diff
@@ -0,0 +1,173 @@
+"""Gradient boosted trees based minimization algorithms."""
+
+import numpy as np
+
+from scipy import stats
+from scipy.optimize import fmin_l_bfgs_b
+from scipy.optimize import OptimizeResult
+
+from sklearn.base import clone
+from sklearn.utils import check_random_state
+
+from .learning import GradientBoostingQuantileRegressor
+from .utils import extract_bounds
+
+
+def _expected_improvement(X, surrogate, y_opt, xi=0.01):
+    """Evaluate expected improvement for `surrogate` model at `x`
+
+    Parameters
+    ----------
+    X : array-like
+        Values at which to evaluate the acquisition function.
+
+    y_opt: float, optional
+        The previous best value on which we want to improve.
+
+    xi: float, default 0.01
+        By how much must a new point improve over the previous best
+        value.
+    """
+    X = np.asarray(X)
+    if X.ndim == 1:
+        X = np.expand_dims(X, axis=0)
+
+    # low and high are assumed to be the 16% and 84% quantiles
+    low, mu, high = surrogate.predict(X).T
```

Comment:
Yes, because reimplementing the acquisition functions is silly. No, because we could do something smarter when you can estimate the quantile directly instead of having to make an assumption about things being gaussian.